source/server/admin/clusters_request.cc

```diff
@@ -0,0 +1,43 @@
+#include "source/server/admin/clusters_request.h"
+
+#include <cstdint>
+#include <memory>
+
+#include "envoy/server/instance.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/http/headers.h"
+#include "source/server/admin/clusters_chunk_processor.h"
+
+#include "clusters_params.h"
+
+namespace Envoy {
+namespace Server {
+
+ClustersRequest::ClustersRequest(uint64_t chunk_limit, Instance& server,
+                                 const ClustersParams& params)
+    : chunk_limit_(chunk_limit), server_(server), params_(params) {}
+
+Http::Code ClustersRequest::start(Http::ResponseHeaderMap& response_headers) {
+  switch (params_.format_) {
+  case ClustersParams::Format::Text:
+    chunk_processor_ = std::make_unique<TextClustersChunkProcessor>(
+        chunk_limit_, response_headers, server_.clusterManager().clusters().active_clusters_);
+    break;
+  case ClustersParams::Format::Json:
+    chunk_processor_ = std::make_unique<JsonClustersChunkProcessor>(
+        chunk_limit_, response_headers, server_.clusterManager().clusters().active_clusters_);
+    response_headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Json);
+    break;
+  default:
+    return Http::Code::BadRequest;
```

Comment:
can we hit this line in a test?

Actually, I think the best practice is not to include a 'default' clause in a switch on an enum. This means the compile will fail if someone adds a new format, which is what you want.

Reply:
Sounds good to me. I'll make the adjustment.